### PR TITLE
Set PowerVS provider_region

### PR DESCRIPTION
**Pull request by jaywcarman**

### CODE REVIEW

File: app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb [hunk 1]
```diff
@@ -118,6 +118,10 @@ def snapshots
     @snapshots ||= snapshots_api.pcloud_cloudinstances_snapshots_getall(cloud_instance_id).snapshots || []
   end
 
+  def provider_region
+    @provider_region = cloud_manager.pcloud_location(connection)
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb:122]: You probably want to ||= this

Comment (jaywcarman) [reply app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb]: Oops, you're right. Just force pushed a new commit with it fixed.

File: spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/refresher_spec.rb [hunk 2]
```diff
@@ -24,9 +24,8 @@
     let(:ems) do
       uid_ems  = Rails.application.secrets.ibm_cloud_power[:cloud_instance_id]
       auth_key = Rails.application.secrets.ibm_cloud_power[:api_key]
-      region   = Rails.application.secrets.ibm_cloud_power[:ibmcloud_region]
```
Comment (agrare) [inline spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/refresher_spec.rb:27]: Can we delete this from config/secrets.default.yml?

File: spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/refresher_spec.rb [hunk 3]
```diff
@@ -106,6 +107,13 @@ def assert_ems_counts
       expect(ems.storage_manager.cloud_volume_types.count).to eq(2)
     end
 
+    def assert_cloud_manager
+      region = Rails.application.secrets.ibm_cloud_power[:ibmcloud_region]
```
Comment (agrare) [inline spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/refresher_spec.rb:111]: Ah nevermind

Comment (jaywcarman) [reply spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/refresher_spec.rb:111]: Yeah, the `ibmcloud_region` value is also used throughout the terraform automation to create our spec environment.

**Comment by agrare** [comment]
I'm kind of surprised this didn't require updating the VCR cassettes...

**Comment by agrare** [comment]
Ah it is because it doesn't issue an API call, it pulls the location out of the connection's Crn header

**Review by agrare — APPROVED** [review]

**Comment by Fryguy** [comment]
Backported to `najdorf` in commit bf4d1b93c964ea499e7ac24c8c69c85e6a9f94f9.

```text
commit bf4d1b93c964ea499e7ac24c8c69c85e6a9f94f9
Author: Adam Grare <adam@grare.com>
Date:   Wed Apr 6 18:52:50 2022 -0400

    Merge pull request #365 from jaywcarman/powervs_provider_region
    
    Set PowerVS provider_region
    
    (cherry picked from commit 268e4ce6d06f06c482d7b44ea933b5e1d331c639)
```

**Pull request merged**